cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler/ArgumentsHandlerTest.kt

```diff
@@ -0,0 +1,202 @@
+                    it.registerLanguage<PythonLanguage>()
+                }
+            assertNotNull(result)
+        }
+    }
+
+    @Test
+    fun testPosOnlyArguments() {
+        var func = result.functions["pos_only_and_args"]
+        assertNotNull(func)
+
+        val list = mapOf("a" to true, "b" to true, "c" to false)
+        list.keys.forEachIndexed { idx, name ->
+            var param = func.parameterEdges.firstOrNull { it.end.name.localName == name }
+            assertNotNull(param, "$name should not be empty")
+            if (list[name] == true) {
+                assertContains(
+                    param.end.modifiers,
+                    PythonLanguage.Companion.MODIFIER_POSITIONAL_ONLY_ARGUMENT
+                )
+            }
+            assertEquals(idx, param.index)
+        }
+    }
+
+    @Test
+    fun testVarargsArguments() {
+        var func = result.functions["test_varargs"]
+        assertNotNull(func, "Function 'test_varargs' should be found")
+
+        var variadicArg = func.parameters["args"]
+        assertNotNull(variadicArg, "Failed to find variadic args")
+        assertEquals(true, variadicArg.isVariadic)
+    }
+
+    @Test
+    fun testKwOnlyArguments() {
+        var func = result.functions["kwd_only_arg"]
+        assertNotNull(func, "Function 'kwd_only_arg' should be found")
+
+        var kwOnlyArg = func.parameters["arg"]
+        assertNotNull(kwOnlyArg, "Failed to find keyword only args")
+        assertContains(kwOnlyArg.modifiers, PythonLanguage.Companion.MODIFIER_KEYWORD_ONLY_ARGUMENT)
+    }
+
+    @Test
+    fun testKwDefaultArguments() {
+        var func = result.functions["kw_defaults"]
+        assertNotNull(func, "Function 'kw_defaults' should be found")
+
+        var kwOnlyParams = mapOf("c" to 2, "d" to null, "e" to 3)
+
+        for ((paramName, expectedDefaultValue) in kwOnlyParams) {
+            var param = func.parameters[paramName]
+            assertNotNull(param, "Failed to find keyword-only argument '$paramName'")
+
+            assertContains(param.modifiers, PythonLanguage.MODIFIER_KEYWORD_ONLY_ARGUMENT)
+
+            if (expectedDefaultValue != null) {
+                assertNotNull(param.default, "Parameter '$paramName' should have a default value")
+                assertEquals(
+                    expectedDefaultValue.toLong(),
+                    param.default?.evaluate(),
+                    "Default value for parameter '$paramName' is incorrect"
+                )
+            } else {
+                assertNull(param.default, "Parameter '$paramName' should not have a default value")
+            }
+        }
+    }
+
+    @Test
+    fun testDefaultsArguments() {
+        var func = result.functions["defaults"]
+        assertNotNull(func, "Function 'defaults' should be found")
+
+        var defaults = mapOf("b" to 1, "c" to 2)
+        for ((paramName, expectedDefaultValue) in defaults) {
```

Comment:
Does it make sense to add a check that there are no additional arguments (e.g. by checking the size of `func.parameters` is as expected)?

Reply:
Done